src/cider/nrepl/middleware/inspect.clj

```diff
@@ -107,16 +110,33 @@
 (defn def-current-value [msg]
   (inspector-response msg (swap-inspector! msg inspect/def-current-value (symbol (:ns msg)) (:var-name msg))))
 
+(defn print-current-value-reply [{:keys [::print/print-fn session] :as msg}]
+  (let [inspector (-> session meta ::inspector)]
+    (with-open [writer (print/replying-PrintWriter :value msg msg)]
+      (binding [*print-length* (or *print-length* 100)
+                *print-level* (or *print-level* 20)]
+        ((or print-fn pprint/pprint) (:value inspector) writer)
+        (.flush writer)))
+    (respond-to msg :status :done)))
+
 (defn tap-current-value [msg]
   (inspector-response msg (swap-inspector! msg inspect/tap-current-value)))
 
 (defn tap-indexed [msg]
   (inspector-response msg (swap-inspector! msg inspect/tap-indexed (:idx msg))))
 
 (defn handle-inspect [handler {:keys [op inspect] :as msg}]
-  (if (and (= op "eval") inspect)
+  (cond
+    (and (= op "eval") inspect)
     (handle-eval-inspect handler msg)
 
+    ;; This is outside of `with-safe-transport` because it streams the pretty
+    ;; printed result to the client in multiple messages. It does NOT return any
+    ;; value, which is expected by `with-safe-transport`.
+    (= op "inspect-print-current-value")
```

Comment:
@alexander-yakushev Should we prefix this with `cider/` from the beginning or leave this to for another day? (e.g. when we add aliases for the all the existing ops)

Reply:
I don't mind; might as well add the prefix now indeed.

Reply:
I would keep it consistent and do it once we decide to rename all. But I can also do this now (we are talking about this op only right?) if you prefer.

Reply:
Okay, let's do this later.